app/src/behaviors/behavior_mod_morph.c

```diff
@@ -84,26 +90,36 @@ static int behavior_mod_morph_init(const struct device *dev) { return 0; }
 
 #define _TRANSFORM_ENTRY(idx, node)                                                                \
     {                                                                                              \
-        .behavior_dev = DEVICE_DT_NAME(DT_INST_PHANDLE_BY_IDX(node, bindings, idx)),               \
-        .param1 = COND_CODE_0(DT_INST_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param1), (0),       \
-                              (DT_INST_PHA_BY_IDX(node, bindings, idx, param1))),                  \
-        .param2 = COND_CODE_0(DT_INST_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param2), (0),       \
-                              (DT_INST_PHA_BY_IDX(node, bindings, idx, param2))),                  \
+        .behavior_dev = DEVICE_DT_NAME(DT_PHANDLE_BY_IDX(node, bindings, idx)),                    \
+        .param1 = COND_CODE_0(DT_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param1), (0),            \
+                              (DT_PHA_BY_IDX(node, bindings, idx, param1))),                       \
+        .param2 = COND_CODE_0(DT_PHA_HAS_CELL_AT_IDX(node, bindings, idx, param2), (0),            \
+                              (DT_PHA_BY_IDX(node, bindings, idx, param2))),                       \
     }
 
-#define KP_INST(n)                                                                                 \
-    static struct behavior_mod_morph_config behavior_mod_morph_config_##n = {                      \
-        .normal_binding = _TRANSFORM_ENTRY(0, n),                                                  \
-        .morph_binding = _TRANSFORM_ENTRY(1, n),                                                   \
-        .mods = DT_INST_PROP(n, mods),                                                             \
-        .masked_mods = COND_CODE_0(DT_INST_NODE_HAS_PROP(n, keep_mods), (DT_INST_PROP(n, mods)),   \
-                                   (DT_INST_PROP(n, mods) & ~DT_INST_PROP(n, keep_mods))),         \
+#define KP_INST(inst)                                                                              \
+    BUILD_ASSERT((COND_CODE_0(DT_NODE_HAS_PROP(inst, binding_params), (0),                         \
+                              ((((DT_PROP_BY_IDX(inst, binding_params, 0) >> 2) &                  \
+                                 (DT_PROP_BY_IDX(inst, binding_params, 0))) |                      \
+                                ((DT_PROP_BY_IDX(inst, binding_params, 1) >> 2) &                  \
+                                 (DT_PROP_BY_IDX(inst, binding_params, 1)))))) == 0),              \
+                 "Invalid binding parameters");                                                    \
```

Comment:
What is this checking exactly? I assume it is checking that `arg1` and `arg2` values are not equal?

If so, it would be nice to make it more explicit, I think it is a bit too clever for me. It can be simplified by at least splitting the assert to two (for binding_params 1 and 2) and a more informative error message can be added.